app/controllers/course_summaries_controller.rb

```diff
@@ -15,4 +15,114 @@ def populate
   def get_marking_scheme_details
     redirect_to url_for(controller: 'marking_schemes', action: 'populate')
   end
+
+  def download_csv_grades_report
+    assignments = Assignment.order(:id)
+    csv_string = CSV.generate do |csv|
+      #Populates the entire CSV with headers, students and their marks
+      populated_csv = populate_students_and_marks(csv)
+    end
+    course_name = "#{COURSE_NAME}"
+    course_name_underscore = course_name.squish.downcase.tr(" ", "_")
+    send_data csv_string, disposition: 'attachment',
+              filename: "#{course_name_underscore}_grades_report.csv"
+  end
+end
+
+#---- HELPER FUNCTIONS FOR DOWNLOAD_CSV_GRADES_REPORT BELOW-----#
+
+#Helper function for download_csv_grades_report. Populates the headers in the CSV
+def populate_header_titles(assignments)
+  header = ['Username']
+  assignments.each do |assignment|
+    header.push(assignment.short_identifier)
+  end
+  header.push("Total marks")
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.